### PR TITLE
fix: show confirm modal when calling a conversation from wrapper top bar menu (WPB-5445)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5445" title="WPB-5445" target="_blank">TaskWPB-5445</a>  [Web] Prevent accidental calls when chosen from top bar menu
  </td></table>
  <br />
 

## Description
clicking on "Call" in the top bar menu of the electron wrapper, while in a group with 4 or more members, will now be prompted with a confirmation modal. This aligns with the behavior already present in the web app when initiating calls within group conversations.

## Checklist

- [X] PR has been self reviewed by the author;
- [X] Hard-to-understand areas of the code have been commented;
- [X] If it is a core feature, unit tests have been added;

### CODE REVIEW

**Comment by PatrykBuniX** [comment]
Hmm, seems like already have this logic defined in CallingViewModel, see https://github.com/wireapp/wire-webapp/blob/fix/clear-content-hides-conversation/src/script/view_model/CallingViewModel.ts#L186. Since we want to check the number of participants every time we try to start a call, maybe this logic could be moved to `CallingRepository.startCall`? WDYT?

File: src/script/calling/CallingRepository.ts [hunk 1]
```diff
@@ -683,16 +685,39 @@ export class CallingRepository {
...
+            text: t('modalConfirmSecondary'),
+          },
+          text: {
+            htmlMessage: `<div class="modal-description">
+            ${t('groupCallConfirmationModalTitle', memberCount)}
+          </div>`,
+            closeBtnLabel: t('groupCallModalCloseBtnLabel'),
+          },
+        });
+      } else {
```
Comment (atomrc) [inline src/script/calling/CallingRepository.ts]: This code must exist somewhere else. We should make sure that the calling button AND the desktop shortcut both go through the exact same code!

Also modals triggering should belong in the view layer (either react component or old view models), not in repositories

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/script/calling/CallingRepository.ts (hunk 1 above)
Comment (thisisamir98) [reply src/script/calling/CallingRepository.ts]: @atomrc I changed the implementation to handle all of the logic in calling view model, could you have another look?

**Review by atomrc — APPROVED** [review]

File: src/script/view_model/CallingViewModel.ts [hunk 2]
```diff
@@ -81,7 +83,7 @@ declare global {
     setSinkId?: (sinkId: string) => Promise<void>;
   }
 }
-const maxGroupSize = 4;
+const MAX_USERS_TO_CALL_WITHOUT_CONFIRM = 1;
```
Comment (V-Gira) [inline src/script/view_model/CallingViewModel.ts]:
```suggestion
const MAX_USERS_TO_CALL_WITHOUT_CONFIRM = 4;
```

Not against having the modal show up every time, but 1 is probably a little low :D

**Review by PatrykBuniX — APPROVED** [review]

File: src/script/view_model/CallingViewModel.ts (hunk 2 above)
Comment (thisisamir98) [reply src/script/view_model/CallingViewModel.ts]: https://github.com/wireapp/wire-webapp/pull/16248/commits/c45196d1e882235a49cc69987d8cf9ed2b5f422f

**Review by V-Gira — APPROVED** [review]

**Pull request merged**